### PR TITLE
Two rows for mutual friendship

### PR DESCRIPTION
**Pull request by dannisonarias**
create a second row for the mutual friendship by reversing the ids and using `ActiveRecord::Base.transaction`
create relative integration test

### CODE REVIEW

**Comment by dannisonarias** [comment]
> ## Changes Required.
> Hi team 
> 
> There are requirements you left out:
> 
> * I am not able to accept a friend request.
>   ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/final_project/milestone5/accept_reject_friendship.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/49736080/))
> * User can reject a friendship request ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/final_project/milestone5/accept_reject_friendship.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/49736080/))
> * Posts timeline includes posts written by logged-in user and their friends ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/final_project/milestone5/friends_timeline.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/49736080/)) However, in your application there is no button to accept friend requests.
> * There should be a button to invite user to friendship on users list page and on single user page ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/final_project/milestone5/user_can_see_invite_buttton.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/49736080/))
> 
> Make these updates, then submit another form for a code review.
> 
> #### `Have a great day` 
> `Uchechukwu Henry Inyama`

> IMPORTANT NOTE: during the code review, you will get hints about implementing mutual friendships in the most efficient way but try to do your best on your own.

After attempting to implement the 2 row friendship some of my code stopped working and this review doesn't address how to implement the 2 row friendship. I will be asking for another code review. Thank you for your review uche-inyama

File: app/models/user.rb [hunk 1]
```diff
@@ -32,9 +38,18 @@ def friend_requests
   end
 
   def confirm_friend(user)
-    friendship = inverse_friendships.find { |i| i.user == user }
+    friendship = inverse_friendships.find { |i| i.user.id == user }
     friendship.confirmed = true
-    friendship.save
+    begin
+      ActiveRecord::Base.transaction do
```
Comment (suretrust) [inline app/models/user.rb:44]: I'm not sure why you need this. Do you really need to catch any error here?

File: app/models/user.rb [hunk 2]
```diff
@@ -14,7 +14,13 @@ class User < ApplicationRecord
   has_many :likes, dependent: :destroy
   has_many :friendships, dependent: :destroy
   has_many :inverse_friendships, class_name: 'Friendship', foreign_key: 'friend_id'
-
+  has_many :confirmed_friendships, -> { where confirmed: true }, class_name: "Friendship"
+  has_many :friends, through: :confirmed_friendships
+  has_many :pending_friendships, -> { where confirmed: nil }, class_name: "Friendship", foreign_key: "user_id"
+  has_many :pending_friends, through: :pending_friendships, source: :friend
+  has_many :inverted_friendships, -> { where confirmed: nil }, class_name: "Friendship", foreign_key: "friend_id"
+  has_many :friend_requests, through: :inverted_friendships, :source => :user
```
Comment (suretrust) [inline app/models/user.rb:22]: `source: :user` is a more recent syntax, hence, it's better.

File: app/models/user.rb (hunk 1 above)
Comment (dannisonarias) [reply app/models/user.rb]: Just to make sure that if there were any errors to occur that both transactions would be rolled back and that way I didn't have a loose solo friendship in the database. I removed this feature, it was mainly there as insurance.

**Pull request merged**